### ISSUE TITLE
Bizarre Behaviour

### ISSUE THREAD

**Issue by IslamHamada** [PierreSenellart/provsql#19]
**Postgres** version: psql (PostgreSQL) 12.4 (Ubuntu 12.4-0ubuntu0.20.04.1)
**Provsql** version: Up till the commit of Oct 7

Some behavior seems inconsistent when I run the following SQL lines:

-- create the first table x1
create table x1 (n integer);
insert into x1 values (10),(20);
select add_provenance('x1');
SELECT create_provenance_mapping('p0', 'x1', 'n');

-- this should work perfectly
select *,formula(provenance(),'p0') from x1;

-- creating table u1 exactly as x1
create table u1 as (select * from x1);
--this command gives the following error "ERROR:  provenance() called on a table without provenance"
select *,formula(provenance(),'p0') from u1;

-- creating another table x2
create table x2 (n integer);
insert into x2 values (30),(40);
select add_provenance('x2');
SELECT create_provenance_mapping('p1', 'x2', 'n');

-- joining the provenance in one table
insert into p0 (select * from p1);

-- creating table u2 as (x1 union x2)
create table u2 as (select * from x1 union select * from x2);
-- this command unexpectedly doesn't give an error. I expect it to produce the same error as u1
select *,formula(provenance(),'p0') from u2;

**The provenance doesn't function properly on table u1 although it works perfectly fine on u2. u1 is exactly like x1, while u2 is the union of x1 and x2.**

**Comment by PierreSenellart** [comment]
Thanks for the bug report. I could reproduce the issue. The difference comes from the type of the `provsql` column in `u1` and `u2`:

```sql
# \d u1
                     Table « public.u1 »
 Colonne |  Type   | Collationnement | NULL-able | Par défaut 
---------+---------+-----------------+-----------+------------
 n       | integer |                 |           | 
 provsql | uuid    |                 |           | 

# \d u2
                          Table « public.u2 »
 Colonne |       Type       | Collationnement | NULL-able | Par défaut 
---------+------------------+-----------------+-----------+------------
 n       | integer          |                 |           | 
 provsql | provenance_token |                 |           | 
```

The correct type should be `provenance_token`, not sure why it is not for `u1`, even though it is in `x1`. I will look into it and fix this.

**Comment by PierreSenellart** [comment]
Investigating further, I could reproduce the issue with PostgreSQL 9.5 and 12, but not with PostgreSQL 11 and 13, which is strange. Looking into it.

**Comment by PierreSenellart** [comment]
This should be fixed in 7414ef0, please fetch the latest commit.
